src/ticks.js

```diff
@@ -2,35 +2,42 @@ const e10 = Math.sqrt(50),
     e5 = Math.sqrt(10),
     e2 = Math.sqrt(2);
 
-function tickSpec(start, stop, count) {
+const logParams = {
+  binary: [Math.log2, 2],
+  natural: [(n) => Math.log1p(n - 1), Math.E],
+  common: [Math.log10, 10],
+}
```

Comment:
> Tiny loss of precision and performance using Math.log(base) etc. instead of Math.LN10

Does this approach work for you to use the respective precise `Math` methods for each allowed log type?